tagstudio/src/qt/widgets/thumb_renderer.py

```diff
@@ -746,6 +757,46 @@ def _video_thumb(self, filepath: Path) -> Image.Image:
             )
         return im
 
+    def _pdf_thumb(self, filepath: Path, size: int) -> Image.Image:
+        file: QFile = QFile(filepath)
+        was_open_success: bool = file.open(
+            QIODeviceBase.OpenModeFlag.ReadOnly, QFileDevice.Permission.ReadUser
+        )
+        if not was_open_success:
+            logging.error(
+                f"[ThumbRenderer][PDF][ERROR]: Couldn't open pdf-file {filepath.name}"
+            )
+            return None
+        document: QPdfDocument = QPdfDocument()
+        document.load(file)
+        # Transform page_size in points to pixels with proper aspect ratio
+        page_size: QSizeF = document.pagePointSize(0)
+        ratio_hw: float = page_size.height() / page_size.width()
+        if ratio_hw >= 1:
+            page_size *= size / page_size.height()
+        else:
+            page_size *= size / page_size.width()
+        # Enlarge image to improve image downscaling (kind of arbitrary)
+        page_size *= 2.5
+        # Render image with no anti-aliasing for speed
+        render_options: QPdfDocumentRenderOptions = QPdfDocumentRenderOptions()
+        render_options.setRenderFlags(
+            QPdfDocumentRenderOptions.RenderFlag.TextAliased
+            | QPdfDocumentRenderOptions.RenderFlag.ImageAliased
+            | QPdfDocumentRenderOptions.RenderFlag.PathAliased
+        )
+        # Convert QImage to PIL Image
+        rendered_qimage: QImage = document.render(0, page_size.toSize(), render_options)
+        buffer: QBuffer = QBuffer()
+        buffer.open(QBuffer.OpenModeFlag.ReadWrite)
+        rendered_qimage.save(buffer, "PNG")
+        pil_image: Image.Image = Image.open(BytesIO(buffer.buffer().data()))
+        buffer.close()
+        # Replace transparent pixels with white (otherwise Background defaults to transparent)
+        pixel_array = np.asarray(pil_image.convert("RGBA")).copy()
+        pixel_array[pixel_array[:, :, 3] == 0] = [255, 255, 255, 255]
```

Comment:
this sounds like something very usefull as its own method. would be useful for others in the future. Can you add this as its own method and just call it here?

Reply:
I am considering creating a utils file for images. But I'm not totally sure how the whole project is laid out.